### PR TITLE
Documented the workflow metadata

### LINKED ISSUE #9475: [Workflow] Add a MetadataStore to fetch some metadata
See https://github.com/symfony/symfony/pull/26092

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #9475 but it's missing an example. Maybe @lyrixx can help us here. Thanks!

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@lyrixx we'll need your review/help to finish this PR (hopefully before 4.1 is released in one month). Thank you!

**Comment by lyrixx** [comment]
OK, I will look at it ASAP

File: workflow/usage.rst [hunk 1]
```diff
@@ -450,3 +450,145 @@ The following example shows these functions in action:
...
+                            ),
+                         ),
+                     ),
+                 ),
+             ),
+         ));
+
+Then, you can access this metadata in your PHP code as follows::
+
+    // MISSING EXAMPLE HERE...
```
Comment (lyrixx) [inline workflow/usage.rst]: You can access it in different ways.

If you are using the workflow directly:

```php
// I executed this code in the controller, so basically I'm fetching a Workflow instance
        $workflow = $this->get('workflow.article');

        $workflow
            ->getMetadataStore()
            ->getWorkflowMetadata()['title'] ?? false
        ;

        $workflow
            ->getMetadataStore()
            ->getPlaceMetadata('draft')['title'] ?? false
        ;

        $aTransition = $workflow->getDefinition()->getTransitions()[0];

        $workflow
            ->getMetadataStore()
            ->getTransitionMetadata($aTransition)['title'] ?? false
        ;

// There is a shortcut that work with everything, see https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Workflow/Metadata/MetadataStoreInterface.php#L34-L36

        $workflow
            ->getMetadataStore()
            ->getMetadata('title')
        ;
```

Use case: Flash message:

```php
            // $transition = ...; (an instance of Transition)
            $title = $this->get('workflow.article')->getMetadataStore()->getMetadata('title', $transition);
            $request->getSession()->getFlashBag()->add('info', "You have successfully applied the transition with title: '$title'");
```


In a listener:

```php
<?php

namespace App\Listener\Workflow\Task;

use Symfony\Component\EventDispatcher\EventSubscriberInterface;
use Symfony\Component\Workflow\Event\GuardEvent;
use Symfony\Component\Workflow\TransitionBlocker;

class DoneGuard implements EventSubscriberInterface
{
    public function guardPublish(GuardEvent $event)
    {
        $timeLimit = $event->getMetadata('time_limit', $event->getTransition());

        if (date('Hi') <= $timeLimit) {
            return;
        }

        $explanation = $event->getMetadata('explaination', $event->getTransition());
        $event->addTransitionBlocker(new TransitionBlocker($explanation , 0));
    }

    public static function getSubscribedEvents()
    {
        return [
            'workflow.task.guard.done' => 'guardPublish',
        ];
    }
}
```

File: workflow/usage.rst [hunk 2]
```diff
@@ -450,3 +450,145 @@ The following example shows these functions in action:
...
+            {% endfor %}
+        </ul>
+    </p>
+    <p>
+        <strong>Enabled transition(s)</strong>
+        <ul>
+            {% for transition in workflow_transitions(article) %}
+                <li>
+                    {{ transition.name }}:
+                    <code>{{ workflow_metadata(article, 'priority', transition) ?: '0' }}</code>
```
Comment (lyrixx) [inline workflow/usage.rst]: Super example !!!

**Review by lyrixx — COMMENTED** [review]
Here we go ;)

I have also done that: https://github.com/lyrixx/SFLive-Paris2016-Workflow/pull/8/files

If you need more help, just ping me ;)

**Review by ndench — APPROVED** [review]

**Comment by alexislefebvre** [comment]
Is this PR ready? Can someone please merge it? Can we help in any way?

**Comment by lyrixx** [comment]
> Is this PR ready? Can someone please merge it? Can we help in any way?

No, There is still a "MISSING EXAMPLE HERE".
If you want you can finish it :heart:

**Comment by alexislefebvre** [comment]
Ok, I'm sorry I thought that [your comment](https://github.com/symfony/symfony-docs/pull/9476#discussion_r184728373) filled this gap.

File: workflow/usage.rst [hunk 3]
```diff
@@ -450,3 +450,145 @@ The following example shows these functions in action:
...
+            // ...
+            'workflows' => array(
+                'blog_publishing' => array(
+                    'metadata' => array(
+                        'title' => 'Blog Publishing Workflow',
+                    ),
+                    // ...
+                    'places' => array(
+                        'draft' => array(
+                            'max_num_of_words' => 500,
```
Comment (noniagriconomie) [inline workflow/usage.rst]: missing metadata array or i miss something? @javiereguiluz

File: workflow/usage.rst (hunk 1 above)
Comment (noniagriconomie) [reply workflow/usage.rst]: nice snippets :)

**Comment by javiereguiluz** [comment]
Closing this in favor of the new PRs created by @pbowyer. Thanks Peter!

**Pull request closed without merging**